dashboard/api/pkg/types/types.go

```diff
@@ -61,18 +55,66 @@ type JobModeration struct {
 	Notes         string    `json:"notes"`
 }
 
-type JobModerationSummary struct {
-	Moderation *JobModeration        `json:"moderation"`
-	Request    *JobModerationRequest `json:"request"`
-	User       *User                 `json:"user"`
+type ModerationRequest interface {
+	GetID() int64
+	GetType() ModerationType
+	GetCallback() *URL
+}
+
+type ModerationSummary struct {
+	Moderation *Moderation       `json:"moderation"`
+	Request    ModerationRequest `json:"request"`
+	User       *User             `json:"user"`
+}
+
+type JobModerationRequest struct {
+	ID       int64          `json:"id"`
+	Created  time.Time      `json:"created"`
+	Callback URL            `json:"callback"`
+	JobID    string         `json:"job_id"`
+	Type     ModerationType `json:"type"`
+}
+
+func (req *JobModerationRequest) GetID() int64 {
+	return req.ID
+}
+
+func (req *JobModerationRequest) GetType() ModerationType {
+	return req.Type
+}
+
+func (req *JobModerationRequest) GetCallback() *URL {
+	return &req.Callback
+}
+
+type JobModerationSummary = ModerationSummary
+
+// A ResultModerationRequest represents a request to moderate a result produced
+// by a job.
+//
+// We explicitly tie the moderation of results to jobs because jobs might be for
+// different users with different trust levels, and also because one job
+// producing a result that is the same as another job might be revealing
+// depending on the execution (i.e. automatically approving a result
+// representing the "empty set" might cause problems). So we tie each moderation
+// request to a moderation request for a job.
+//
+// However, we do not require every result from the same job to be moderated if
+// the results are the same (and we can detect that e.g. using a CID). If the
+// `metadata` field of the StorageSpec contains a `hash` key, we will also use
+// this to identify duplicate results.
```

Comment:
```suggestion
// However, we do not require every result from the same job to be moderated if
// the previously moderated results are the same (and we can detect that e.g. 
// using a CID). If the `metadata` field of the StorageSpec contains a `hash` key, 
// we will also use this to identify duplicate results.
```

Is it worth clarifying that the description applies to previously moderated results,
and not just any job with the same results?

Reply:
Well the idea is that if a job uses `--concurrency=3` and generates 3 identical results, one should only have to moderate these once. But ideally this should mean that when receiving 3 VerifierRequests for the same hash, only 1 moderation request should be created. 

But the code doesn't actually do this currently and isn't even well structured to do it (atm it assumes one-to-one between execution ID and moderation request). So I think I will just take this comment out, and add an issue for adding this in later.